architecture.tex

```diff
@@ -53,58 +55,190 @@
 
 Python allows classes to overload operators. The Python interpreter translates
-above \texttt{x*y + 2} to, roughly,
+the above \texttt{x*y + 2} to, roughly,
 \verb|(x.__mul__(y)).__add__(2)|. \texttt{x} and \texttt{y}, returned from
 the \texttt{symbols} function, are \texttt{Symbol} instances. The \texttt{2}
 in the expression is processed by Python as a literal, and is stored as
 Python's builtin \texttt{int} type. When \texttt{2} is called by the
-\verb|__add__|, it is converted to the SymPy type \texttt{Integer(2)}. In
+\verb|__add__| method, it is converted to the SymPy type \texttt{Integer(2)}. In
 this way, SymPy expressions can be built in the natural way using Python
 operators and numeric literals.
 
 One must be careful in one particular instance. Python does not have a builtin
-rational literal type. Given a fraction of integers in \texttt{1/2}, Python
-will perform floating point division and produce \texttt{0.5}\footnote{This is
-  the behavior in Python 3. In Python 2, \texttt{1/2} will perform integer
-  division and produce \texttt{0}, unless one uses \texttt{from \_\_future\_\_
-    import division}}. Because Python uses eager evaluation, expressions like
-\texttt{x + 1/2} will produce \texttt{x + 0.5}, because by the time a SymPy
-function sees the \texttt{1/2} it has already been converted to \texttt{0.5}
-by Python. However, for a CAS like SymPy, we typically want to work with exact
-rational numbers whenever possible. Working around this is simple, however:
-one can wrap one of the integers with \texttt{Integer}, like \texttt{x +
-  Integer(1)/2}, or using \texttt{x + Rational(1, 2)}. This gotcha is a small
-downside to using Python directly instead of a custom DSL, and we consider it
-to be worth it for the advantages listed above.
-
-\subsection{The Core}
+rational literal type. Given a fraction of integers such as \texttt{1/2},
+Python will perform floating point division and produce
+\texttt{0.5}~\footnote{This is the behavior in Python 3. In Python 2,
+  \texttt{1/2} will perform integer division and produce \texttt{0}, unless
+  one uses \texttt{from \_\_future\_\_ import division}.}. Python uses eager
+evaluation, so expressions like \texttt{x + 1/2} will produce \texttt{x +
+  0.5}, and by the time any SymPy function sees the \texttt{1/2} it has
+already been converted to \texttt{0.5} by Python. However, for a CAS like
+SymPy, one typically wants to work with exact rational numbers whenever
+possible. Working around this is simple, however:  one can wrap one of the
+integers with \texttt{Integer}, like \texttt{x + Integer(1)/2}, or using
+\texttt{x + Rational(1, 2)}. SymPy provides a function \texttt{S} which can be
+used to convert objects to SymPy types with minimal typing, such as \texttt{x
+  + S(1)/2}. This gotcha is a small downside to using Python
+directly instead of a custom DSL, and we consider it to be worth it for the
+advantages listed above.
```

Comment:
I was thinking about that. I think our preferred way for people to use SymPy interactively should be to use the Jupyter notebook. It is possible to do this there as well, by passing a flag to `init_session`, but, I just tested, and it doesn't seem to work in the notebook.